CAIPs/caip-25.md

```diff
@@ -16,422 +16,229 @@ CAIP-25 defines an authorization procedure for a chain agnostic provider to inte
-
-The successful result MUST contain an object called `sessionScopes` which contains 1 or more `scopeObjects`.
-- All required `scopeObjects` and all, none, or some of the optional `scopeObject`s (at the discretion of the provider) MUST be included if successful.
-- Unlike the request, each scope object MUST also contain an `accounts` array,
-containing 0 or more [CAIP-10][]-conformant accounts authorized for the session
-and valid in that scope. Additional constraints on the accounts authorized for a given session MUST be applied conformant to the namespace's [CAIP-10][] profile, if one has been specified.
-
-A `scopedProperties` object MAY also be present, each member of which corresponds to exactly 1 `sessionScope`.
-This is intended for expressing connection-specific or non-standardized extensions to `sessionScope`.
-Each object in `scopedProperties` MUST be keyed to a `scopeString`, and SHOULD correspond to a `sessionScopes` entry with the same key.
-There are no type, depth, or shape constraints on the contents of each property in `scopedProperties`.
-If an object in `scopedProperties` is keyed to a `scopeString` not currently authorized for the session, it SHOULD be ignored.
-
-A `sessionProperties` object MAY also be present, with no protocol-wide shape constraints or semantics assumed.
-This is intended for expressing metadata about the CAIP-25 connection and accessible to ALL `sessionScopes` equally.
-There are no type, depth, or shape constraints on the contents of `sessionProperties`.
-
-An example of a successful response follows:
-
 ```jsonc
 {
   "id": 1,
   "jsonrpc": "2.0",
   "result": {
     "sessionId": "0xdeadbeef",
-    "sessionScopes": {
-      "eip155": {
-        "references": ["1", "137"],
-        "methods": ["eth_sendTransaction", "eth_signTransaction", "get_balance", "eth_sign", "personal_sign"]
+    "scopes": {
+      "wallet:any": {
+        "methods": ["wallet_authenticate", "wallet_pay"],
+        "notifications": [],
+        "capabilities": {}
+      },
+      "wallet:eip155": {
+        "methods": ["wallet_grantPermissions", "wallet_getAssets"],
+        "notifications": [],
+        "capabilities": {
+          "walletService": "https://wallet-service.example.com/rpc"
+        }
+      },
+      "eip155:1": {
+        "accounts": ["0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"],
+        "methods": ["eth_sendTransaction", "personal_sign"],
         "notifications": ["accountsChanged", "chainChanged"],
-        "accounts": ["eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb", "eip155:137:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"]
+        "capabilities": {}
```

Comment:
I left this comment somewhere before, but `accounts` should be required in each scopeObject here... with the possible exception of the new `wallet:any` scope

Reply:
oh wait nevermind I see they are here now, just moved... I'm just being dense. Do we need to specify somewhere how/why the `wallet:*` scopes don't need accounts though?

Reply:
couldn't they be a required array but empty array is a valid value? that feels more explicit, like "I AM GIVING YOU ZERO ACCOUNTS FOR THIS" 😅 

(and, sidenote, might a wallet namespace be the right place to stick an ephemeral/implicit/offchain-only address used for completely off-node usecases like presenting verifiable credentials? a girl can dream...)